### PR TITLE
added a gamemap class

### PR DESCRIPTION
**Pull request by jarradbattaglia**
Created a item class
created create item in entityfactory
have automatic moving in movement system (need to add controls from networking)
print out map when running (every 10 seconds)
put map in networking instead fo Data.map

### CODE REVIEW

File: src/main/scala/com/ayai/systems/MovementSystem.scala [hunk 1]
```diff
@@ -18,8 +19,19 @@ class MovementSystem(a: Aspect = Aspect.getAspectFor(classOf[Position]) ) extend
       override def process(e: Entity) {
         var p: Position = positionMapper.get(e)
         var velocity : Velocity = velocityMapper.get(e)
-        var test : Int = world.delta.intValue()
-        p.x += velocity.speed*world.delta
-        p.y += velocity.speed*world.delta
+        var test : Int = world.delta.intValue
+        //move position over one
+        if(p.x+1 < 0 ||  p.x+1 > map.width || p.y+1 < 0 || p.y+1 > map.height) {
```
Comment (timothyhahn) [inline src/main/scala/com/ayai/systems/MovementSystem.scala]: Can you simply just not this whole statement?

Comment (jarradbattaglia) [inline src/main/scala/com/ayai/systems/MovementSystem.scala]: fixed

File: src/main/scala/com/ayai/maps/GameMap.scala [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+		var oldPosition : Position = getPosition(id)
+		coordByEnitity.remove(id)
+		coordByEnitity.put(id,position)
+		entityByCoord(oldPosition.x)(oldPosition.y) = -1
+		entityByCoord(position.x)(position.y) = id
+	}
+
+	//get position of entity
+	def getPosition(id : Int) : Position = {
+		return coordByEnitity(id)
```
Comment (timothyhahn) [inline src/main/scala/com/ayai/maps/GameMap.scala]: Remove last line returns please. I count 6 in this commit.

Comment (jarradbattaglia) [inline src/main/scala/com/ayai/maps/GameMap.scala]: yep as i said didnt know you didnt need to do it 

For anyone reading over this, last line is returned

**Pull request merged**